# HW1

In [1]:
import pandas as pd

# Prob 1

In [2]:
# Prob 1

data_table1 = pd.read_csv("table1.txt", sep='\t')

# Prob 1.1

In [3]:
# Prob1.1
import statsmodels.api as sm

# Extracting the response variable y and predictor variable x8
y = data_table1['y']
x8 = data_table1['X8']

# Adding a constant for intercept term
x8 = sm.add_constant(x8)

# Fitting the linear regression model
model = sm.OLS(y, x8).fit()

model.params

const    21.788251
X8       -0.007025
dtype: float64

# Prob 1.2

In [4]:
# Prob 1.2
# Constructing the ANOVA table

model.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                      y   R-squared:                       0.545
Model:                            OLS   Adj. R-squared:                  0.527
Method:                 Least Squares   F-statistic:                     31.10
Date:                Sun, 29 Oct 2023   Prob (F-statistic):           7.38e-06
Time:                        19:43:13   Log-Likelihood:                -63.123
No. Observations:                  28   AIC:                             130.2
Df Residuals:                      26   BIC:                             132.9
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const         21.7883      2.696      8.081      0.000      16.246      27.330
X8            -0.0070      0.001     -5.577      0.000      -0.010      -0.004
==============================================================================
Omnibus:                        2.076   Durbin-Watson:                   1.566
Prob(Omnibus):                  0.354   Jarque-Bera (JB):                1.402
Skew:                           0.305   Prob(JB):                        0.496
Kurtosis:                       2.089   Cond. No.                     1.28e+04
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 1.28e+04. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

# Prob 1.3

In [5]:
# Prob 1.3
# Extracting the 95% confidence interval for the slope (x8 coefficient)
slope_ci = model.conf_int().loc['X8']

slope_ci

0   -0.009614
1   -0.004436
Name: X8, dtype: float64

# Prob 1.4
# The answer to this question is provided by the $R^2$ value, which was found to be 0.545 or 54.5% from the regression output.

# Prob 1.5

In [6]:
# Prob 1.5
# Predicting the mean number of games won for x8 = 2000 with a 95% confidence interval
prediction = model.get_prediction(exog=[[1, 2000]])
ci_mean = prediction.conf_int()

ci_mean

array([[6.76575309, 8.71034757]])

------

# Prob 2

In [7]:
# Prob 2

data_table2 = pd.read_csv("table2.txt", sep='\t')
data_table2.head()

y      x1     x2     x3     x4     X5
0  271.8  783.35  33.53  40.55  16.66  13.20
1  264.0  748.45  36.50  36.19  16.46  14.11
2  238.8  684.45  34.66  37.31  17.66  15.68
3  230.7  827.80  33.13  32.52  17.50  10.53
4  251.6  860.45  35.75  33.71  16.40  11.00

# Prob 2.1

In [8]:
# Extracting the response variable y and predictor variable x4
y_heat_flux = data_table2['y']
x4_radial_deflection = data_table2['x4']

# Adding a constant for intercept term
x4_radial_deflection = sm.add_constant(x4_radial_deflection)

# Fitting the linear regression model
model_heat_flux = sm.OLS(y_heat_flux, x4_radial_deflection).fit()

model_heat_flux.params

const    607.103265
x4       -21.402458
dtype: float64

# Prob 2.2

In [9]:
model_heat_flux.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                      y   R-squared:                       0.721
Model:                            OLS   Adj. R-squared:                  0.710
Method:                 Least Squares   F-statistic:                     69.61
Date:                Sun, 29 Oct 2023   Prob (F-statistic):           5.94e-09
Time:                        19:44:45   Log-Likelihood:                -112.96
No. Observations:                  29   AIC:                             229.9
Df Residuals:                      27   BIC:                             232.6
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const        607.1033     42.906     14.150      0.000     519.067     695.139
x4           -21.4025      2.565     -8.343      0.000     -26.666     -16.139
==============================================================================
Omnibus:                        0.520   Durbin-Watson:                   0.963
Prob(Omnibus):                  0.771   Jarque-Bera (JB):                0.362
Skew:                          -0.260   Prob(JB):                        0.834
Kurtosis:                       2.828   Cond. No.                         315.
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

# Prob 2.3

In [10]:
# Extracting the 99% confidence interval for the slope (x4 coefficient)
slope_ci_99 = model_heat_flux.conf_int(alpha=0.01).loc['x4']

slope_ci_99

0   -28.509951
1   -14.294965
Name: x4, dtype: float64

# Prob 2.4

In [11]:
# Extracting the R^2 value from the model
r_squared = model_heat_flux.rsquared

r_squared

0.7205241964067103

# Prob 2.5

In [12]:
# Predicting the mean heat flux for x4 = 16.5 with a 95% confidence interval
prediction_heat_flux = model_heat_flux.get_prediction(exog=[[1, 16.5]])
ci_mean_heat_flux = prediction_heat_flux.conf_int()

ci_mean_heat_flux

array([[249.14675201, 258.77865527]])